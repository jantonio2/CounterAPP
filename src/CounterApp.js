import PropTypes from 'prop-types';
import { useState } from 'react';

const CounterApp = ({value = 10}) => {

  const [counter, setCounter] = useState(value);

  // handleAdd
  const handleAdd = () => {
    setCounter(counter + 1);
    // setCounter( (c) => c + 1 );
  };

  // handleReset
  const handleReset = () => {
    setCounter(value);
  };

  // handleReset
  const handleSubtract = () => {
    setCounter(counter - 1);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2> { counter } </h2>
     {/* Cuando ponemos () nos saldra undefined al renderizar. Se esta pasando como referencia */}
      <button onClick={handleAdd}>Sumar +1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleSubtract}>Restar -1</button>
    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number
}

CounterApp.defaultProps = {
  value: 10
}

export default CounterApp;