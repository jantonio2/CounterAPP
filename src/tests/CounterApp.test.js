import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Pruebas en <CounterApp />', () => {
  test('debe de mostrar <CounterApp /> correctamente', () => {
    const wrapper = shallow(<CounterApp />);

    expect(wrapper).toMatchSnapshot();
  });
  
  test('debe de mostrar el valor enviado', () => {
    const wrapper = shallow(
      <CounterApp
        value = {100}
      />
    );

    const textoH2 = parseFloat(wrapper.find('h2').text().trim());

    expect(textoH2).toBe(100);
  })
  
})
