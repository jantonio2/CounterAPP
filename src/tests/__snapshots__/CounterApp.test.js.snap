// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe de mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
    Sumar +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    Restar -1
  </button>
</Fragment>
`;
